openmeter/billing/service/lineservice/meters.go

```diff
@@ -0,0 +1,131 @@
+package lineservice
+
+import (
+	"context"
+	"fmt"
+	"slices"
+
+	"github.com/alpacahq/alpacadecimal"
+
+	billingentity "github.com/openmeterio/openmeter/openmeter/billing/entity"
+	"github.com/openmeterio/openmeter/openmeter/productcatalog/feature"
+	"github.com/openmeterio/openmeter/openmeter/streaming"
+	"github.com/openmeterio/openmeter/pkg/models"
+)
+
+type getFeatureUsageInput struct {
+	Line       *billingentity.Line
+	ParentLine *billingentity.Line
+	Meter      models.Meter
+	Feature    feature.Feature
+	Subjects   []string
+}
+
+func (i getFeatureUsageInput) Validate() error {
+	if i.Line == nil {
+		return fmt.Errorf("line is required")
+	}
+
+	if i.Line.ParentLineID != nil && i.ParentLine == nil {
+		return fmt.Errorf("parent line is required for split lines")
+	}
+
+	if i.Line.ParentLineID == nil && i.ParentLine != nil {
+		return fmt.Errorf("parent line is not allowed for non-split lines")
+	}
+
+	if i.ParentLine != nil {
+		if i.ParentLine.Status != billingentity.InvoiceLineStatusSplit {
+			return fmt.Errorf("parent line is not split")
+		}
+	}
+
+	if slices.Contains([]models.MeterAggregation{
+		models.MeterAggregationAvg,
+		models.MeterAggregationMin,
+	}, i.Meter.Aggregation) {
+		if i.ParentLine != nil {
+			return fmt.Errorf("aggregation %s is not supported for split lines", i.Meter.Aggregation)
+		}
+	}
+
+	if len(i.Subjects) == 0 {
+		return fmt.Errorf("subjects are required")
+	}
+
+	return nil
+}
+
+type featureUsageResponse struct {
+	// LinePeriodQty is the quantity of the usage for the line for the period
+	LinePeriodQty alpacadecimal.Decimal
+	// PreLinePeriodQty is the quantity of the usage for the line for the period before the current period
+	PreLinePeriodQty alpacadecimal.Decimal
+}
+
+func (s *Service) getFeatureUsage(ctx context.Context, in getFeatureUsageInput) (*featureUsageResponse, error) {
+	// Validation
+	if err := in.Validate(); err != nil {
+		return nil, err
+	}
+
+	meterQueryParams := &streaming.QueryParams{
+		Aggregation:   in.Meter.Aggregation,
+		FilterSubject: in.Subjects,
+		From:          &in.Line.Period.Start,
+		To:            &in.Line.Period.End,
```

Comment:
feature has group by filters that should be applied here too

Reply:
It's already included here: https://github.com/openmeterio/openmeter/pull/1787/files#diff-12d9d74ba62a6275562925d58ede10dfb9708c723129ef2b114e2053efc63473R79-R84